pkg/kubelet/cm/topologymanager/socketmask/socketmask.go

```diff
@@ -39,7 +39,13 @@ type SocketMask interface {
 
 type socketMask uint64
 
-//NewSocketMask creates a new SocketMask
+// NewEmptySocketMask creates a new, empty SocketMask
+func NewEmptySocketMask() SocketMask {
```

Comment:
Could you provide some examples here to prove the newly added functions like `NewEmptySocketMask` etc would do a great job at clean up the code

Reply:
Sure.

As a side note, I wrote the original `socketmask` abstraction as well as the `TopologyManager` code referenced below (even though someone else ultimately committed them). I'm just coming in after it now to clean it up before the 1.17 release.

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/topologymanager/topology_manager.go#L154
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/topologymanager/topology_manager.go#L169

The code above can be simplified from:
```
	defaultAffinity, _ := socketmask.NewSocketMask()
	defaultAffinity.Fill()
```
to:
```
	defaultAffinity := socketmask.NewEmptySocketMask().Fill()
```

I'd also be OK, having the original `NewSocketMask()` panic instead of returning an error, but I don't like that everywhere in the codebase (including future code in the PRs referenced below) where we currently create `SocketMask`, we end up ignoring the error anyway.

Likewise, with the introduction of the package level `And` function, the code at:
https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/topologymanager/topology_manager.go#L200

Can be simplified from:
```
		mergedAffinity, _ := socketmask.NewSocketMask()
		mergedAffinity.Fill()
		mergedAffinity.And(socketAffinities...)
```
to:
```
		mergedAffinity := socketmask.And(socketAffinities...)
```

There are more places in the code in pending PRs that will benefit from this as well:
https://github.com/kubernetes/kubernetes/pull/73920
https://github.com/kubernetes/kubernetes/pull/74345

The plan is to create a cleanup PR for all of these places once those other PRs have merged.

Reply:
Sounds Good